components-compose/src/main/java/uk/gov/hmrc/components/compose/organism/summaryrow/SummaryRowView.kt

```diff
@@ -0,0 +1,142 @@
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.CompositionLocalProvider
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.painter.Painter
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.semantics.Role
+import androidx.compose.ui.semantics.focused
+import androidx.compose.ui.semantics.role
+import androidx.compose.ui.semantics.semantics
+import androidx.compose.ui.text.style.TextOverflow
+import uk.gov.hmrc.components.compose.R
+import uk.gov.hmrc.components.compose.ui.theme.HmrcRippleTheme
+import uk.gov.hmrc.components.compose.ui.theme.HmrcTheme.dimensions
+import uk.gov.hmrc.components.compose.ui.theme.HmrcTheme.typography
+
+object SummaryRowView {
+
+    @Composable
+    operator fun invoke(
+        modifier: Modifier = Modifier,
+        titleText: String,
+        isBoldTitleTextAppearance: Boolean = true,
+        titleMaxLines: Int = -1,
+        icon: Painter = painterResource(id = R.drawable.components_ic_chevron_right),
+        chevronContentDescription: String = "",
+        accessibilityMessage: String? = null,
+        rows: List<@Composable () -> Unit>,
+        readerTrait: ReaderTrait = ReaderTrait.READER_TRAIT_INFO,
+        onSummaryRowClicked: (() -> Unit)? = null
+    ) {
+        if (onSummaryRowClicked != null) {
+            CompositionLocalProvider(LocalRippleTheme provides HmrcRippleTheme) {
+                Row(
+                    modifier = Modifier.clickable { onSummaryRowClicked.invoke() },
+                    verticalAlignment = Alignment.CenterVertically
+                ) {
+                    SummaryRow(
+                        modifier = modifier.then(
+                            Modifier
+                                .padding(
+                                    top = dimensions.hmrcSpacing16,
+                                    bottom = dimensions.hmrcSpacing16,
+                                    start = dimensions.hmrcSpacing16
+                                )
+                                .weight(1f)
+                        ),
+                        titleText = titleText,
+                        titleMaxLines = titleMaxLines,
+                        isBoldTitleTextAppearance = isBoldTitleTextAppearance,
+                        readerTrait = readerTrait,
+                        rows = rows
+                    )
+                    Spacer(modifier = Modifier.width(dimensions.hmrcSpacing8))
+                    Image(
+                        painter = icon,
+                        modifier = Modifier
+                            .semantics { role = Role.Button }
+                            .clickable { onSummaryRowClicked() },
+                        contentDescription = chevronContentDescription
+                    )
+                    Spacer(modifier = Modifier.width(dimensions.hmrcSpacing16))
```

Comment:
Does this need to be clickable and a button if the whole row is clickable already? Will this result in some sort of double click?
If this does need to be clickable, is the touch state large enough for accessibility? Should the surrounding spacers be included in the Image modifier as padding rather than added separately?

Also, the space on the right of the icon looks a bit large currently, I think we can reduce this (maybe to 8dp) to make more space for the text and so it doesn't look so indented - I think the icon image has some spacing around it.

Reply:
That clickable introduced because of the accessibility announcement which is aligned with previous sample example.